.github/workflows/py-publish.yml

```diff
@@ -29,6 +31,11 @@ jobs:
     - name: Check distribution
       run: twine check dist/*
 
+    - name: Create attestations
+      uses: actions/attest-build-provenance@v1
+      with:
+        subject-path: 'dist/*'
```

Comment:
From my understanding, we want a single digest for each type of distributed wheel, yes? I guess that [using wildcards here (doc link)](https://github.com/actions/attest-build-provenance#identify-subjects-by-wildcard) will generate a digest for each file/artifact that satisfies the glob pattern. 👍🏼

> [!NOTE]
> In this workflow, only a bdist (binary distribution) is built. This should be fine for most use cases since all of our python packages are written in pure python.
> 
> If a user tries to install one of our packages from source using [`--no-binary`](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-no-binary)
> ```
> pip install --no-binary cpp-linter
> ```
> then they might get an error about the release not having a sdist (source distribution) asset to build from source.